src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -460,6 +461,48 @@ void reset() {
   @VisibleForTesting
   abstract int getTempRowCount();
 
+  abstract void closeInternal();
+
+  /**
+   * Close the row buffer and release resources. Note that the caller needs to handle
+   * synchronization
```

Comment:
Would it be difficult to make it synchronized? Why not mark the function as `synchronized` since it will probably be called infrequently?

Reply:
ack, I made `close()` synchronized.